Components/CategoryResolver/DefaultCategoryResolver.php

```diff
@@ -100,7 +112,26 @@ public function storeRemoteCategories(array $categories, $articleId)
             $productToCategory->setConnectCategory($remoteCategory);
             $this->manager->persist($productToCategory);
         }
+    }
 
-        $this->manager->flush();
+    /**
+     * @param \Shopware\CustomModels\Connect\RemoteCategory[] $assignedCategories
+     * @param $articleId
+     */
+    private function removeProductsFromRemoteCategory(array $assignedCategories, $articleId)
+    {
+        $currentProductCategories = $this->productToRemoteCategoryRepository->getArticleRemoteCategories($articleId);
+
+        $assignedCategoryIds = array_map(function (RemoteCategory $assignedCategory) {
+            $assignedCategory->getId();
+        }, $assignedCategories);
+
+        /** @var ProductToRemoteCategory $currentProductCategory */
+        foreach ($currentProductCategories as $currentProductCategory) {
+            if (!in_array($currentProductCategory->getConnectCategoryId(), $assignedCategoryIds)) {
+                //DELETE PRODUCT ASSIGNMENT
+                $this->manager->remove($currentProductCategory);
```

Comment:
would be much less load if you just use delete DQL/SQL here - maybe as a repository method?